src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js

```diff
@@ -0,0 +1,129 @@
+import {useEffect} from 'react';
+import PropTypes from 'prop-types';
+import lodashGet from 'lodash/get';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../../ONYXKEYS';
+import * as ReportUtils from '../../ReportUtils';
+import reportPropTypes from '../../../pages/reportPropTypes';
+import {withNavigationPropTypes} from '../../../components/withNavigation';
+import * as App from '../../actions/App';
+import usePermissions from '../../../hooks/usePermissions';
+import CONST from '../../../CONST';
+import Navigation from '../Navigation';
+
+const propTypes = {
+    /** Available reports that would be displayed in this navigator */
+    reports: PropTypes.objectOf(reportPropTypes),
+
+    /** The policies which the user has access to */
+    policies: PropTypes.objectOf(
+        PropTypes.shape({
+            /** The policy name */
+            name: PropTypes.string,
+
+            /** The type of the policy */
+            type: PropTypes.string,
+        }),
+    ),
+
+    isFirstTimeNewExpensifyUser: PropTypes.bool,
+
+    /** Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        /** Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /** If the admin room should be opened */
+            openOnAdminRoom: PropTypes.bool,
+
+            /** The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    ...withNavigationPropTypes,
+};
+
+const defaultProps = {
+    reports: {},
+    policies: {},
+    isFirstTimeNewExpensifyUser: false,
+};
+
+/**
+ * Get the most recently accessed report for the user
+ *
+ * @param {Object} reports
+ * @param {Boolean} [ignoreDefaultRooms]
```

Comment:
Does not seem to have a default value set, also there are required params coming after
```suggestion
 * @param {Boolean} ignoreDefaultRooms
```

Reply:
I'm not sure I understand. This is just an internal function and doesn't require a default value, it should always be called with a boolean value:

![CleanShot 2023-09-27 at 08 15 37](https://github.com/Expensify/App/assets/1634213/500ac697-223a-4c6e-ada8-dfe40f075356)

Reply:
Yeah but the `[]` in js docs means optional parameter, but this is not optional parameter but the looks of things

Reply:
Ah, this wasn't my code, just moved it from the wrapper. But it is fixed.